hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/qjournal/client/QuorumJournalManager.java

```diff
@@ -618,9 +620,10 @@ private void selectStreamingInputStreams(
     Map<AsyncLogger, RemoteEditLogManifest> resps =
         loggers.waitForWriteQuorum(q, selectInputStreamsTimeoutMs,
             "selectStreamingInputStreams");
-
-    LOG.debug("selectStreamingInputStream manifests:\n" +
-        Joiner.on("\n").withKeyValueSeparator(": ").join(resps));
+    if (LOG.isDebugEnabled()) {
+      LOG.debug("selectStreamingInputStream manifests:\n" +
+          Joiner.on("\n").withKeyValueSeparator(": ").join(resps));
+    }
```

Comment:
In this case, I don't think there is a simple way to avoid the evaluation without the `if (LOG.isDebugEnabled())`, so this one is good.